system-addon/common/PrerenderData.jsm

```diff
@@ -1,32 +1,80 @@
-const prefConfig = {
-  // Prefs listed with "invalidates: true" will prevent the prerendered version
+class _PrerenderData {
+  constructor(options) {
+    this.initialPrefs = options.initialPrefs;
+    this.initialSections = options.initialSections;
+    this._setValidation(options.validation);
+  }
+
+  get validation() {
+    return this._validation;
+  }
+
+  set validation(value) {
+    this._setValidation(value);
+  }
+
+  get invalidatingPrefs() {
+    return this._invalidatingPrefs;
+  }
+
+    // This is needed so we can use it in the constructor
+  _setValidation(value = []) {
+    this._validation = value;
+    this._invalidatingPrefs = value.reduce((result, next) => {
+      if (typeof next === "string") {
+        result.push(next);
+        return result;
+      } else if (next && next.oneOf) {
+        return result.concat(next.oneOf);
+      }
+      throw new Error("Your validation configuration is not properly configured");
+    }, []);
+  }
+
+  arePrefsValid(getPref) {
+    for (const name of this.validation) {
+      if (typeof name === "string") {
+        if (getPref(name) !== this.initialPrefs[name]) {
+          return false;
+        }
+      } else if (name && name.oneOf) {
```

Comment:
I suppose you could just make each item an array. A 1 item array means it must match. A multi-item array means any of them can match.

Reply:
I feel like have the `oneOf` makes it more immediately obvious what the behaviour is though when you're just reading the file